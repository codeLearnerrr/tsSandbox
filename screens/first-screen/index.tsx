import React, { useState } from 'react'
import {
    View,
    Text,
    Image,
    KeyboardAvoidingView,
    Button
} from 'react-native'
import {
    AddOperation,
    SubtractOperation,
    DivideOperation,
    MultiplyOperation
} from '../../components/operations'

import randomImage from '../../assets/image.jpg';
import { TextInput } from 'react-native-gesture-handler';


export const ShowSomething = ({ navigation }: any) => {
    const [firstValue, setFirstValue] = useState<number>(0);
    const [secondValue, setSecondValue] = useState<number>(0);

    return (
        <View style={{ flex: 1, alignItems: 'center' }}>
            {/* <Button title='go to second screen' onPress={() => navigation.navigate('NotHome')} /> */}

            < Image
                resizeMethod='resize'
                resizeMode='contain'
                source={randomImage}
                style={{ width: 100, height: 100, alignSelf: 'center', borderRadius: 10, marginTop: 10 }
                }
            />

            <Text style={{ fontSize: 30, marginBottom: 30, textAlign: 'center' }}>
                Type numbers in{'\n'} each grey box to calculate:
                </Text>

            <View style={{ flex: 1, flexDirection: 'column', maxHeight: 100, }}>
                <View style={{ flex: 1, flexDirection: 'row' }}>
                    <TextInput placeholder='' keyboardType='numeric' style={{
                        borderColor: 'grey',
                        borderBottomWidth: 1,
                        backgroundColor: 'rgba(0,0,0,.1)',
                        maxHeight: 70,
                        maxWidth: 70,
                        width: 70,
                        borderRadius: 8,
                        fontSize: 20,
                        textAlign: 'center',
                    }}
                        onChangeText={(input) => setFirstValue(parseInt(input))}
                    />
                    <Text style={{ fontSize: 15, marginHorizontal: 15 }}>{''}</Text>
                    <TextInput placeholder='' keyboardType='numeric' style={{
                        borderColor: 'grey',
                        borderBottomWidth: 1,
                        backgroundColor: 'rgba(0,0,0,.1)',
                        maxHeight: 70,

                        maxWidth: 70,
                        width: 70,
                        borderRadius: 8,
                        fontSize: 20,
                        textAlign: 'center',
                    }}
                        onChangeText={(input) => setSecondValue(parseInt(input))}
                    />

                </View>

            </View>

            <View style={{ flex: 1, maxHeight: 200, maxWidth: 300, }}>

                <Text numberOfLines={1} style={{ fontSize: 25, }}>Addition: {AddOperation({ firstNumber: firstValue, secondNumber: secondValue })}</Text>
                <Text numberOfLines={1} style={{ fontSize: 25, }}>Subtraction: {SubtractOperation({ firstNumber: firstValue, secondNumber: secondValue })} </Text>
                <Text numberOfLines={1} style={{ fontSize: 25, }}>Division: {DivideOperation({ firstNumber: firstValue, secondNumber: secondValue })} </Text>
                <Text numberOfLines={1} style={{ fontSize: 25, }}>Multiplication: {MultiplyOperation({ firstNumber: firstValue, secondNumber: secondValue })} </Text>

            </View>

        </View>
    )
}

