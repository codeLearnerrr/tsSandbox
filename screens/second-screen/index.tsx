import React from 'react'
import {
    View,
    Text,
    Image,
    Button
} from 'react-native'

import randomImage from '../../assets/image.jpg';

export const ShowSomething2 = ({ navigation }: any) => {
    return (
        <>
            <Button title='go to second screen' onPress={() => navigation.navigate('Home')} />

            <Image resizeMethod='resize' resizeMode='contain' source={randomImage} style={{ width: 50, height: 50 }} />
            <View style={{ flex: 1, justifyContent: 'center', alignItems: 'center' }}>
                <Text>
                    Second screen here
                </Text>
            </View>
        </>
    )
}


