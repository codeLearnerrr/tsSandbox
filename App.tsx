/**
 * React native calculator using typescript
 * https://github.com/facebook/react-native
 *
 * @format
 * @flow strict-local
 */

import React from 'react';

import { NavigationContainer } from '@react-navigation/native';
import { createStackNavigator } from '@react-navigation/stack';

import { ShowSomething } from './screens/first-screen';
import { ShowSomething2 } from './screens/second-screen';

const Stack = createStackNavigator();

const App: () => React$Node = () => {
  return (
    <>      
      <NavigationContainer>
        <Stack.Navigator headerMode='none' initialRouteName='Home'>
          <Stack.Screen name="Home" component={ShowSomething} />
          <Stack.Screen name="NotHome" component={ShowSomething2} />
        </Stack.Navigator>
      </NavigationContainer>
    </>
  );
};

export default App;
