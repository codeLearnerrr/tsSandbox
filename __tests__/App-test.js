/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';



import { ShowSomething } from '../screens/first-screen';
import { ShowSomething2 } from '../screens/second-screen';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  renderer.create(<App />);
});

it('renders correctly ShowSomething', () => {
  const tree = renderer.create(<ShowSomething />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly ShowSomething2', () => {
  const tree = renderer.create(<ShowSomething2 />).toJSON();
  expect(tree).toMatchSnapshot();
});

