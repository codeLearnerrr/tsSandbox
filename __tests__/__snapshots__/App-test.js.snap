// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly ShowSomething 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        go to second screen
      </Text>
    </View>
  </View>
  <Image
    resizeMethod="resize"
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/image.jpg",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "borderRadius": 10,
        "height": 100,
        "marginTop": 30,
        "width": 100,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 30,
        "marginBottom": 30,
        "textAlign": "center",
      }
    }
  >
    Type numbers in
    

     each grey box to calculate:
  </Text>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "column",
        "maxHeight": 100,
        "minHeight": 100,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        collapsable={false}
        keyboardType="numeric"
        onChangeText={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        placeholder=""
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.1)",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderRadius": 8,
            "fontSize": 20,
            "maxHeight": 70,
            "maxWidth": 70,
            "textAlign": "center",
            "width": 70,
          }
        }
        underlineColorAndroid="transparent"
      />
      <Text
        style={
          Object {
            "fontSize": 15,
            "marginHorizontal": 15,
          }
        }
      >
        
      </Text>
      <TextInput
        allowFontScaling={true}
        collapsable={false}
        keyboardType="numeric"
        onChangeText={[Function]}
        onGestureHandlerEvent={[Function]}
        onGestureHandlerStateChange={[Function]}
        placeholder=""
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,.1)",
            "borderBottomWidth": 1,
            "borderColor": "grey",
            "borderRadius": 8,
            "fontSize": 20,
            "maxHeight": 70,
            "maxWidth": 70,
            "textAlign": "center",
            "width": 70,
          }
        }
        underlineColorAndroid="transparent"
      />
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "maxHeight": 200,
        "maxWidth": 300,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Addition: 
      0
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Subtraction: 
      0
       
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Division: 
      NaN
       
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "fontSize": 25,
        }
      }
    >
      Multiplication: 
      0
       
    </Text>
  </View>
</View>
`;

exports[`renders correctly ShowSomething2 1`] = `
Array [
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        go to second screen
      </Text>
    </View>
  </View>,
  <Image
    resizeMethod="resize"
    resizeMode="contain"
    source={
      Object {
        "testUri": "../../../assets/image.jpg",
      }
    }
    style={
      Object {
        "height": 50,
        "width": 50,
      }
    }
  />,
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Second screen here
    </Text>
  </View>,
]
`;
